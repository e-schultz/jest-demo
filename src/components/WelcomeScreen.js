import React, { useState, useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import axios from 'axios'
import styled from '@emotion/styled'

import * as actionTypes from '../store/actions'
import Spinner from './Spinner'

const StyledApp = styled.div`
    text-align: center;
`

const StyledHeader = styled.div`
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
`

const StyledTitle = styled.h1`
    font-size: 1.5em;
`

const WelcomeScreen = () => {
    const [fetchedPhones, setFetchedPhones] = useState([])
    const [loading, setLoading] = useState(false)

    const dispatch = useDispatch()

    useEffect(() => {
        setLoading(true)
        axios
            .get('//www.mocky.io/v2/5918bc6b120000701040dbec')
            .then(response => {
                setFetchedPhones(response.data.phones)
                setLoading(false)
            })
            .catch(error => {
                console.log(error)
            })
    }, [])

    return (
        <StyledApp
            className="container d-flex align-items-center justify-content-center"
            style={{ height: '100vh' }}
        >
            <div className="row">
                <div className="col-md-12 text-center">
                    <StyledHeader>
                        <StyledTitle>Welcome to Phone Catalog</StyledTitle>
                    </StyledHeader>
                </div>
                <div className="col-md-12 text-center">
                    <Link
                        onClick={() =>
                            dispatch({
                                type: actionTypes.ADD_PHONES,
                                payload: fetchedPhones,
                            })
                        }
                        to="/phones"
                        className="btn btn-primary"
                        style={
                            loading
                                ? {
                                      pointerEvents: 'none',
                                      cursor: 'not-allowed',
                                      opacity: '0.2',
                                  }
                                : {
                                      pointerEvents: 'all',
                                      cursor: 'pointer',
                                      opacity: '1',
                                  }
                        }
                    >
                        Go to catalog
                    </Link>
                </div>
                {loading && <Spinner />}
            </div>
        </StyledApp>
    )
}

export default WelcomeScreen
