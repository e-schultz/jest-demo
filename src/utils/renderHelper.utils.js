import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { render as rtlRender } from '@testing-library/react'
import reducer from '../store/reducer'

// you can provide initialState or the entire store that the ui is rendered with
export const renderHelper = (
    ui,
    {
        initialState,
        store = createStore(reducer, initialState),
        ...renderOptions
    } = {}
) => {
    function Wrapper({ children }) {
        return (
            // MemoryRouter enables us to test Routes
            <MemoryRouter>
                <Provider store={store}>{children}</Provider>
            </MemoryRouter>
        )
    }
    return {
        ...rtlRender(ui, {
            wrapper: Wrapper,
            ...renderOptions,
        }),
        // adding `store` to the returned utilities to allow us
        // to reference it in our tests (just try to avoid using
        // this to test implementation details).
        store,
    }
}
