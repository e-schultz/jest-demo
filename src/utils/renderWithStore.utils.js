import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render as rtlRender } from '@testing-library/react'
import reducer from '../store/reducer'

// you can provide initialState or the entire store that the ui is rendered with
export const renderWithStore = (
    ui,
    {
        initialState,
        store = createStore(reducer, initialState),
        ...renderOptions
    } = {}
) => {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return {
        ...rtlRender(ui, {
            wrapper: Wrapper,
            ...renderOptions,
        }),
        // adding `store` to the returned utilities to allow us
        // to reference it in our tests (just try to avoid using
        // this to test implementation details).
        store,
    }
}
