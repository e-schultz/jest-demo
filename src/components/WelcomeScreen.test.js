import WelcomeScreen from './WelcomeScreen'

import { renderWithStore } from '../utils/renderWithStore.utils'

const initialState = {
    phones: [],
}

describe('WelcomeScreen should render correctly', () => {
    it('should match snapshot', () => {
        const { container } = renderWithStore(<WelcomeScreen />, {
            initialState,
        })
        expect(container.firstChild).toMatchSnapshot()
    })
})
