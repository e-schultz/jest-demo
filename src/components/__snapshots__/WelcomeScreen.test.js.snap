// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should render correctly should match snapshot 1`] = `
<div
  class="container d-flex align-items-center justify-content-center css-18ul79m"
  style="height: 100vh;"
>
  <div
    class="row"
  >
    <div
      class="col-md-12 text-center"
    >
      <div
        class="css-cp5izk"
      >
        <h1
          class="css-12yjyrl"
        >
          Welcome to Phone Catalog
        </h1>
      </div>
    </div>
    <div
      class="col-md-12 text-center"
    >
      <div
        class="css-myp9de"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
